### PR TITLE
[TRTLLM-10362][feat] Added Mamba and MLA layers to the sharding tests

### LINKED ISSUE #8595: [AutoDeploy:] update file list for multi-gpu test trigger
### 🚀 The feature, motivation and pitch

Multi-gpu tests in PRs are only triggered if the diff includes files from the following list:
https://github.com/NVIDIA/TensorRT-LLM/blob/9b3d7cc3e64d046d11098e7b52020699a67e156b/jenkins/L0_MergeRequest.groovy#L660-L726

Right now, the only `auto_deploy`-related files in the list are the multi-gpu test files themselves. It is missing any source-code files from `tensorrt_llm/_torch/auto_deploy` that could affect multi-gpu tests. 

Todo: update file list to include `auto_deploy` source code files for better multi-gpu test coverage.

Relevant threads:
1. https://nvidia.slack.com/archives/C059LSY62BT/p1761164693295449?thread_ts=1761152375.823869&cid=C059LSY62BT
2. https://nvidia.slack.com/archives/C08T55LHSG4/p1761163648138149

### Alternatives

_No response_

### Additional context

_No response_

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and checked the [documentation](https://nvidia.github.io/TensorRT-LLM/) and [examples](https://github.com/NVIDIA/TensorRT-LLM/tree/main/examples) for answers to frequently asked questions.

### PR DESCRIPTION
**Pull request by greg-kwasniewski1**
Fixes #10362 
Fixes #8595 

## Release Notes

* **Tests**
  * Extended multi-GPU sharding tests to cover additional model architectures and tensor parallelism patterns.
  * Added test coverage for new model variants in auto-deploy sharding validation.

* **Chores**
  * Updated integration test configuration to trigger multi-GPU workflows for newly covered model types.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--reuse-test (optional)pipeline-id --disable-fail-fast --skip-test --stage-list "A10-PyTorch-1, xxx" --gpu-type "A30, H100_PCIe" --test-backend "pytorch, cpp" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx" --detailed-log --debug(experimental)]`

Launch build/test pipelines. All previously running jobs will be killed.

`--reuse-test (optional)pipeline-id ` *(OPTIONAL)* : Allow the new pipeline to reuse build artifacts and skip successful test stages from a specified pipeline or the last pipeline if no pipeline-id is indicated. If the Git commit ID has changed, this option will be always ignored. The DEFAULT behavior of the bot is to reuse build artifacts and successful test results from the last pipeline.

`--disable-reuse-test ` *(OPTIONAL)* : Explicitly prevent the pipeline from reusing build artifacts and skipping successful test stages from a previous pipeline. Ensure that all builds and tests are run regardless of previous successes.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-PyTorch-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-PyTorch-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--test-backend "pytorch, cpp"` *(OPTIONAL)* : Skip test stages which don't match the specified backends. Only support [pytorch, cpp, tensorrt, triton]. Examples: "pytorch, cpp" (does not run test stages with tensorrt or triton backend). Note: Does **NOT** update GitHub pipeline status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests in addition to running L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx".

`--detailed-log ` *(OPTIONAL)* : Enable flushing out all logs to the Jenkins console. This will significantly increase the log volume and may slow down the job.

`--debug ` *(OPTIONAL)* : **Experimental feature**. Enable access to the CI container for debugging purpose. Note: Specify exactly one stage in the `stage-list` parameter to access the appropriate container environment. Note: Does **NOT** update GitHub check status.

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`
and the `scripts/test_to_stage_mapping.py` helper.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

### CODE REVIEW

**Review by tcherckez-nvidia — APPROVED** [review]
LGTM

File: tests/integration/defs/accuracy/test_llm_api_autodeploy.py [hunk 1]
```diff
@@ -218,10 +218,12 @@ def test_bf16(self):
             task.evaluate(llm)
 
     @pytest.mark.skip_less_device_memory(32000)
-    def test_fp8(self):
+    @pytest.mark.parametrize("world_size", [1, 4])
```
Comment (2ez4bz) [inline tests/integration/defs/accuracy/test_llm_api_autodeploy.py]: Nit: is there any way we could mark `4` as `pytest.mark.skipif` when the available world size is not large enough?

File: tests/unittest/_torch/auto_deploy/unit/multigpu/transformations/library/test_tp_sharding.py [hunk 2]
```diff
@@ -125,6 +134,75 @@ def forward(self, x):
         return self.linear2(y)
 
 
+class MLA_Block(nn.Module):
```
Comment (2ez4bz) [inline tests/unittest/_torch/auto_deploy/unit/multigpu/transformations/library/test_tp_sharding.py]: Just out of curiosity - was this change related to the mamba issue, or "just" added for more test coverage? If the former, could you explain how it affected the mamba mixer layer?

**Review by 2ez4bz — APPROVED** [review]

File: tests/integration/defs/accuracy/test_llm_api_autodeploy.py (hunk 1 above)
Comment (lucaslie) [reply tests/integration/defs/accuracy/test_llm_api_autodeploy.py]: those need to be configured individually in the `test-db` anyway (see file diff below). So I think it's fine

**Review by lucaslie — APPROVED** [review]

File: tests/unittest/_torch/auto_deploy/unit/multigpu/transformations/library/test_tp_sharding.py (hunk 2 above)
Comment (greg-kwasniewski1) [reply tests/unittest/_torch/auto_deploy/unit/multigpu/transformations/library/test_tp_sharding.py]: The latter: MLA test was added just for more test coverage. Unrelated to Mamba.

File: tests/integration/defs/accuracy/test_llm_api_autodeploy.py
Comment (lucaslie) [inline tests/integration/defs/accuracy/test_llm_api_autodeploy.py]: TP>1 most likely won't pass since multi-stream+TP doesn't work yet, cc @suyoggupta

File: tests/integration/test_lists/test-db/l0_dgx_h100.yml [hunk 3]
```diff
@@ -126,7 +126,10 @@ l0_dgx_h100:
   - disaggregated/test_auto_scaling.py::test_minimal_instances[http-round_robin]
   - disaggregated/test_auto_scaling.py::test_disagg_server_restart[http-round_robin]
   # ------------- AutoDeploy tests ---------------
-  - accuracy/test_llm_api_autodeploy.py::TestNemotronMOE::test_bf16
+  - accuracy/test_llm_api_autodeploy.py::TestNemotronMOE::test_bf16[1]
+  - accuracy/test_llm_api_autodeploy.py::TestNemotronMOE::test_bf16[4]
+  - accuracy/test_llm_api_autodeploy.py::TestNemotronMOE::test_fp8[1]
+  - accuracy/test_llm_api_autodeploy.py::TestNemotronMOE::test_fp8[4]
```
Comment (lucaslie) [inline tests/integration/test_lists/test-db/l0_dgx_h100.yml]: shouldn't test single-gpu on a multi-gpu node. Do single-gpu in the single-gpu stages (`l0_h100.yml`)

**Comment by greg-kwasniewski1** [comment]
/bot run --extra-stage "DGX_B200-4_GPUs-AutoDeploy-1, DGX_H100-4_GPUs-AutoDeploy-1"

**Comment by greg-kwasniewski1** [comment]
/bot run --extra-stage "DGX_B200-4_GPUs-AutoDeploy-1, DGX_H100-4_GPUs-AutoDeploy-1"

**Comment by greg-kwasniewski1** [comment]
/bot run --extra-stage "DGX_B200-4_GPUs-AutoDeploy-1, DGX_H100-4_GPUs-AutoDeploy-1"

**Comment by greg-kwasniewski1** [comment]
/bot run --extra-stage "DGX_B200-4_GPUs-AutoDeploy-1, DGX_H100-4_GPUs-AutoDeploy-1"

**Comment by greg-kwasniewski1** [comment]
/bot run --extra-stage "DGX_B200-4_GPUs-AutoDeploy-1, DGX_H100-4_GPUs-AutoDeploy-1"

**Comment by greg-kwasniewski1** [comment]
/bot run --extra-stage "DGX_B200-4_GPUs-AutoDeploy-1, DGX_H100-4_GPUs-AutoDeploy-1"

**Comment by greg-kwasniewski1** [comment]
/bot run --extra-stage "DGX_B200-4_GPUs-AutoDeploy-1, DGX_H100-4_GPUs-AutoDeploy-1"

**Comment by greg-kwasniewski1** [comment]
/bot run --extra-stage "DGX_B200-4_GPUs-AutoDeploy-1, DGX_H100-4_GPUs-AutoDeploy-1"

**Review by chzblych — APPROVED** [review]
